lucene/highlighter/src/java/org/apache/lucene/search/uhighlight/FieldOffsetStrategy.java

```diff
@@ -193,13 +193,13 @@ protected void createOffsetsEnumsForAutomata(Terms termsIndex, int doc, List<Off
     }
 
     for (int i = 0; i < automata.length; i++) {
-      CharArrayMatcher automaton = automata[i];
+      LabelledCharArrayMatcher automaton = automata[i];
       List<PostingsEnum> postingsEnums = automataPostings.get(i);
       if (postingsEnums.isEmpty()) {
         continue;
       }
-      // Build one OffsetsEnum exposing the automata.toString as the term, and the sum of freq
-      BytesRef wildcardTerm = new BytesRef(automaton.toString());
+      // Build one OffsetsEnum exposing the automaton label as the term, and the sum of freq
```

Comment:
I wonder if we should stop calling these things "automata" and now call them "matcher"?  even if just in some method like perhaps this one.  Automata is a bit of an implementation detail.